### ISSUE TITLE
Explicit version pinning for internal dependencies

### ISSUE THREAD

**Issue by ianmetcalf** [aws/aws-xray-sdk-node#269]
It is currently not possible to downgrade to a lower patch version of the full SDK.

For instance, when the following is set in `package.json`
```
  "dependencies": {
    "aws-xray-sdk": "3.0.0-alpha.1"
  }
```
The `3.0.0-alpha.2` versions of `aws-xray-sdk-core`, `aws-xray-sdk-express`, `aws-xray-sdk-mysql`, and `aws-xray-sdk-postgres` are still installed and used.

**Comment by willarmiros** [comment]
This is a good callout, it's because the full SDK uses the semver carrot `^` when specifying its [dependencies](https://github.com/aws/aws-xray-sdk-node/blob/master/packages/full_sdk/package.json). This should be dropped to make the dependency versions exact. Note this change will only apply to future versions.

**Comment by ianmetcalf** [comment]
Totally understood about only applying to future versions.

The immediate work around for us is to explicitly pin everything in `package.json`
```
  "dependencies": {
    "aws-xray-sdk": "3.0.0-alpha.1",
    "aws-xray-sdk-core": "3.0.0-alpha.1",
    "aws-xray-sdk-express": "3.0.0-alpha.1",
    "aws-xray-sdk-mysql": "3.0.0-alpha.1",
    "aws-xray-sdk-postgres": "3.0.0-alpha.1"
  }
```

**Comment by willarmiros** [comment]
This fix won't require a manual code change, instead we will change the way we publish to use lerna's [`--exact`](https://github.com/lerna/lerna/tree/master/commands/version#--exact) option starting with the next release.

**Comment by willarmiros** [comment]
This issue is resolved as of 3.1.0.
